### PR TITLE
Use built in log rotation capabilities of the docker daemon

### PR DESCRIPTION
**Pull request by faraazkhan**
This better aligns kops with the approach that GCI seems to be taking and resolves: https://github.com/kubernetes/kops/issues/526. For a better explanation see: https://github.com/kubernetes/kubernetes/pull/40634

Watch https://asciinema.org/a/468j319f9hidtak0fl8sm2ekn to see this PR in action. 

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2095)

### CODE REVIEW

**Comment by chrislovecnm** [comment]
@k8s-bot ok to test

**Comment by chrislovecnm** [comment]
Can we add a unit test here please https://github.com/kubernetes/kops/blob/master/nodeup/pkg/model/docker_test.go for the nodeup changes.

Thanks!

**Comment by faraazkhan** [comment]
@chrislovecnm can you explain what you'd like to me to test in the model and how it is relevant to this PR since it changes nothing in: https://github.com/kubernetes/kops/blob/master/nodeup/pkg/model/docker.go . The only change here is in the docker config and it seems that populate_cluster is the only place that stuff is tested in?

File: pkg/apis/kops/dockerconfig.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ type DockerConfig struct {
 	LogLevel *string `json:"logLevel,omitempty" flag:"log-level"`
 	IPTables *bool   `json:"ipTables,omitempty" flag:"iptables"`
 	IPMasq   *bool   `json:"ipMasq,omitempty" flag:"ip-masq"`
+	LogOpts  *string `json:"logOpts,omitempty" flag:"log-opt"`
```
Comment (chrislovecnm) [inline pkg/apis/kops/dockerconfig.go]: Can we make `LogOpt` to match the way the other variables are plural vs singular?

**Comment by faraazkhan** [comment]
made requested changed @chrislovecnm

**Comment by chrislovecnm** [comment]
e2e is building ...

**Comment by sstarcher** [comment]
@faraazkhan can we have the default value of the log opts match Kubernetes - https://github.com/kubernetes/kubernetes/pull/40634/files#diff-6e3b476b9225d1213dc6ad13e453fc16R482

max-size=10m
max-file=5

**Comment by justinsb** [comment]
Yes - this is great , thank you @faraazkhan.  It looks like LogOpt is actually a `[]string`.  I feel we should also make LogDriver at the same time as well.  I'm not entirely sure whether a `[]string` flag can be marshalled today.

It's also going to be tricky _not_ changing this for k8s 1.4 / 1.5...

**Comment by justinsb** [comment]
Thanks @faraazkhan ... going to merge and then apply the changes we need for 1.6!

**Pull request merged**